client/config/cert-and-project-map.ts

```diff
@@ -800,65 +803,60 @@ function getJavaScriptAlgoPath(project: string) {
     : `${jsAlgoBase}/${project}`;
 }
 
-const certsWithoutFullStack = [...legacyCerts, ...certs] as const;
-
-const liveCerts = [...certsWithoutFullStack, legacyFullStack] as const;
+type FilteredCert<T, U> = T extends { certSlug: U } ? T : never;
 
-export type CertsToProjects = Record<
-  (typeof certs)[number]['title'],
-  (typeof certs)[number]['projects']
+type CurrentCert = FilteredCert<
+  (typeof allStandardCerts)[number],
+  (typeof currentCertifications)[number]
 >;
 
-export type LegacyCertsToProjects = Record<
-  (typeof legacyCerts)[number]['title'],
-  (typeof legacyCerts)[number]['projects']
+type LegacyCert = FilteredCert<
+  (typeof allStandardCerts)[number],
+  (typeof legacyCertifications)[number]
 >;
 
-export type UpcomingCertsToProjects = Record<
-  (typeof upcomingCerts)[number]['title'],
-  (typeof upcomingCerts)[number]['projects']
+type UpcomingCert = FilteredCert<
+  (typeof allStandardCerts)[number],
+  (typeof upcomingCertifications)[number]
 >;
 
-const certsToProjects = certs.reduce((acc, curr) => {
-  return {
-    ...acc,
-    [curr.title]: curr.projects
-  };
-}, {} as CertsToProjects);
+const currentCerts = allStandardCerts.filter((cert): cert is CurrentCert =>
+  currentCertifications.includes(cert.certSlug)
+);
+const legacyCerts = allStandardCerts.filter((cert): cert is LegacyCert =>
+  legacyCertifications.includes(cert.certSlug)
+);
+const upcomingCerts = allStandardCerts.filter((cert): cert is UpcomingCert =>
+  upcomingCertifications.includes(cert.certSlug)
+);
```

Comment:
The rationale here is that the groups of certifications (current, legacy and upcoming) should be defined once, but they were implicitly defined twice (once here because they appearand once in certification-settings). 

What I've tried to do here is keep the definitions of certification groups (current, legacy and upcoming) as simple as possible - i.e. arrays. That meant there were implicitly two sources of truth for those groups, since they were also defined here.  

This filtering is my attempt to square this particular circle. Combined with some basic tests, it should be hard to mess up or misuse the config.